engine/crates/composition/src/subgraphs/keys.rs

```diff
@@ -97,12 +139,39 @@ impl<'a> KeyWalker<'a> {
 }
 
 impl<'a> DefinitionWalker<'a> {
+    pub fn is_entity(self) -> bool {
+        self.entity_keys().next().is_some()
+            || self
+                .subgraphs
+                .keys
+                .nested_key_fields
+                .objects_with_nested_keys
+                .contains(&self.id)
```

Comment:
This stuff is ripe for some apis on a bit higher level, so you don't need to traverse the tree so deep to ask that question.

Reply:
It's nested because we're defining methods on `Subgraphs`, but we're in the module that defines NestedKeyFields.